plugins/cr8-copy-s3/src/main/java/io/crate/copy/s3/S3FileOutputFactory.java

```diff
@@ -23,13 +23,30 @@
 
 import io.crate.execution.engine.export.FileOutput;
 import io.crate.execution.engine.export.FileOutputFactory;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.function.Function;
 
 public class S3FileOutputFactory implements FileOutputFactory {
 
     public static final String NAME = "s3";
+    static final Setting<String> PROTOCOL = new Setting<>(
```

Comment:
Same here regarding enum and error message.

Maybe share the protocol setting definition between input and output?